src/explorer/CosmosDBTreeItem.ts

```diff
@@ -38,87 +38,70 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
                 label: 'Install Cosmos DB Extension...'
             })];
         }
-        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
-        const connections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
-        // tslint:disable-next-line:strict-boolean-expressions
-        const unit = connections.find((x: IConnections) => x.webAppId === this.root.client.id) || <IConnections>{};
-        if (!unit.cosmosDB || unit.cosmosDB.length === 0) {
-            return [new GenericTreeItem(this, {
-                commandId: 'appService.AddCosmosDBConnection',
-                contextValue: 'AddCosmosDBConnection',
-                label: 'Add Cosmos DB Connection...'
-            })];
+
+        if (ext.cosmosAPI === undefined) {
+            ext.cosmosAPI = <VSCodeCosmosDB>cosmosDB.exports;
         }
-        return unit.cosmosDB.map(connectionId => {
-            return new CosmosDBDatabase(this, connectionId);
+
+        const mongoAppSettingsKeys: string[] = [];
+        // tslint:disable-next-line:strict-boolean-expressions
+        const appSettings = (await this.root.client.listApplicationSettings()).properties || {};
+        Object.keys(appSettings).forEach((key) => {
+            if (/^mongo/i.test(appSettings[key])) {
```

Comment:
Mmm maybe not quite this simple. I think you should use the regexp you previously had for just the prefix: `^mongodb[^\/]*:\/\/`

Although `^\/` seems like it should be `^:` to me since ':' comes before '/' in the connection string.